web/viewer.css

```diff
@@ -1758,24 +1829,23 @@ html[dir="rtl"] #documentPropertiesOverlay .row > * {
   }
   /* wrapper around (scaled) print canvas elements */
   #printContainer > div {
+    /* Without the following max-height declaration, Chromium might create extra
+    * blank pages, even though it shouldn't! */
+    max-height: 100%;
     page-break-after: always;
     page-break-inside: avoid;
 
-    /* The wrapper always cover the whole page. */
-    height: 100%;
-    width: 100%;
-
+    /* If we're smaller than the page, center the canvas horizontally and
+    * vertically */
+    min-height: 100%;
+    min-width: 100%;
     display: flex;
     flex-direction: column;
     justify-content: center;
     align-items: center;
   }
   #printContainer canvas,
   #printContainer img {
-    /* The intrinsic canvas / image size will make sure that we fit the page. */
-    max-width: 100%;
-    max-height: 100%;
-
```

Comment:
These changes look first of all unrelated to *this* patch, and secondly they're basically reverting a bunch of recent printing fixes; please undo these changes.